### ISSUE TITLE
Entity metadata for * was not found

### ISSUE THREAD

**Issue by prolink007** [typeorm/typeorm#381]
When trying to create the schema for my tables i am getting the following error.

`error: Entity metadata for Comic#variants was not found.`

```
@Entity()
@TableInheritance("class-table")
@DiscriminatorColumn({name: "type", type: "string"})
export class BaseComic {

    @PrimaryGeneratedColumn()
    public id: number;

}
```

```
@ClassEntityChild()
export class Comic extends BaseComic {

    @OneToMany(type => Variant, variant => variant.comic)
    public variants: Variant[];
}
```

```
@ClassEntityChild()
export class Variant extends BaseComic {

    @ManyToOne(type => Comic, comic => comic.variants)
    public comic: Comic;
}
```

What am i doing wrong? I left out some other fields that are irrelevant.

Thanks

**Comment by prolink007** [comment]
Found the issue. I was not adding the entities to the entities configuration.

**Comment by NoNameProvided** [comment]
For future reference: another reasons for this error are discussed in #420.

**Comment by NoNameProvided** [comment]
@prolink007 your issue is solved right? You may want to close the issue then

**Comment by tamannaizap** [comment]
am getting this error.. i dnt know the reason,, Installed all of the dependencies,please help  

SyntaxError: src/entity/photo.js: Unexpected token (3:0)
  1 | import {Entity, Column, PrimaryGeneratedColumn} from "typeorm";
  2 |  
> 3 | @Entity()
    | ^
  4 | export class Photo {
  5 |  
  6 |     @PrimaryGeneratedColumn()
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! trial@1.0.0 start: `babel src --out-dir dist && node dist/index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the trial@1.0.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/dixit/.npm/_logs/2017-11-15T09_55_34_519Z-debug.log

**Comment by disperse** [comment]
For me, this error was thrown because I forgot to add the `@Entity` decorator at the top of my class file. 

Make sure both classes on either side of the relationship are declared as entities.

@tamanna-makkar This is an old issue but you need to make sure to capitalize `Entity` in your decorator.

**Comment by jelling** [comment]
How about this error:

```
EntityMetadataNotFound: No metadata for \"Function\" was found.\n    at new EntityMetadataNotFoundError (/Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/error/EntityMetadataNotFoundError.js:19:28)\n    at Connection.getMetadata (/Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/connection/Connection.js:351:19)\n    at /Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/persistence/EntityPersistExecutor.js:105:80\n    at Array.forEach (<anonymous>)\n    at EntityPersistExecutor.<anonymous> (/Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/persistence/EntityPersistExecutor.js:100:54)\n    at step (/Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/persistence/EntityPersistExecutor.js:32:23)\n    at Object.next (/Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/persistence/EntityPersistExecutor.js:13:53)\n    at /Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/persistence/EntityPersistExecutor.js:7:71\n    at new Promise (<anonymous>)\n    at __awaiter (/Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/persistence/EntityPersistExecutor.js:3:12)\n    at /Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/persistence/EntityPersistExecutor.js:92:108\n    at Array.map (<anonymous>)\n    at EntityPersistExecutor.<anonymous> (/Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/persistence/EntityPersistExecutor.js:92:75)\n    at step (/Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/persistence/EntityPersistExecutor.js:32:23)\n    at Object.next (/Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/persistence/EntityPersistExecutor.js:13:53)\n    at /Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/persistence/EntityPersistExecutor.js:7:71\n    at new Promise (<anonymous>)\n    at __awaiter (/Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/persistence/EntityPersistExecutor.js:3:12)\n    at /Users/jonhaven/Dev/Maverick/mav-web-orm/server/node_modules/typeorm/persistence/EntityPersistExecutor.js:76:60
```
I do not have an entity called Function so my gut is that this is something internal to TypeORM. 

It happens when trying to save this entity:

```

import { Loan } from "./loan";
import { PrimaryGeneratedColumn, Column, ManyToOne, OneToMany, Entity, RelationId, JoinColumn } from "typeorm";
import { MetaFields } from "./metaFields";
import { Customer } from "./customer";
import { SoloK } from "./solok";
import { Bank } from "./bank";

@Entity()
export class PlanParticipant {
    @PrimaryGeneratedColumn() id: number;

    @ManyToOne(type => SoloK, { eager: true })
    soloK: SoloK;

    @RelationId((participant: PlanParticipant) => participant.soloK)
    planId: number;

    @ManyToOne(type => Customer)
    // @JoinColumn({ referencedColumnName: "userId" })
    customer: Promise<Customer>;

    @RelationId((participant: PlanParticipant) => participant.customer)
    personId: string;

    @Column(type => Bank)
    bank: Bank = new Bank();

    @OneToMany(type => Loan, Loan => Loan.participant, { eager: true })
    loans: Loan[];

    @Column(type => MetaFields)
    meta: MetaFields;
}

```

**Comment by DevelGitH** [comment]
And for me this error was thrown because Visual Studio Code automatically imported the other entity, but with the wrong file path. Make sure that the lower- and uppercase of the file path is correct.

**Comment by JaffParker** [comment]
I have this error, but it's definitely not because of wrong import paths or a missing Entity decorator. I went to Connection.buildMetadatas added a console.log to see how they're detected after this line (in the built code):

```js
var entityMetadatas = connectionMetadataBuilder.buildEntityMetadatas(this.options.entities || []);
```

**All my entities were there.** Then I go to load a specific resource and that function ends up being called again, but with only a handful of all my entities, thus breaking with this error. It breaks specifically at relations. I've tried removing relations that it errors at one by one, but then what's the point of relations at all? Should I just get rid of all the relations now?

I can't make a minimal reproduction repo as it would require exposing a lot of our DB structure, but to begin, why is it rebuilding metadatas on a request?
